script/check_changelog.py

```diff
@@ -0,0 +1,131 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Small script to check the changelog. Used by 'changelog.yml' and pre-commit.
+
+If no issue number is provided we only check that proper formatting is respected."""
+
+from __future__ import annotations
+
+import argparse
+import re
+import sys
+from collections.abc import Iterator
+from pathlib import Path
+from re import Pattern
+
+ISSUE_NUMBER_PATTERN = re.compile(r"#\d{1,5}")
+VALID_CHANGELOG_PATTERN: Pattern[str] = re.compile(
+    r"(\*\s[\S[\n ]+?]*\n\n\s\s(Refs|Closes|Follow-up in|Fixes part of)) (PyCQA/astroid)?#\d{1,5}"
+)
+VALID_ISSUE_NUMBER_PATTERN: Pattern[str] = re.compile(r"\*[\S\s]*?#\d{1,5}")
+DOC_PATH = (Path(__file__).parent / "../doc/").resolve()
+PATH_TO_WHATSNEW = DOC_PATH / "whatsnew"
+UNCHECKED_VERSION = [
+    # Not checking version prior to 1.0.0 because the issues referenced are a mix
+    # between Logilab internal issue and Bitbucket. It's hard to tell, it's
+    # inaccessible for Logilab and often dead links for Bitbucket anyway.
+    # Not very useful generally, unless you're an open source historian.
+    "0.x",
+    # Too much Bitbucket issues in this one :
+    "1.0",
+    "1.1",
+    "1.2",
+]
+
+NO_CHECK_REQUIRED_FILES = {
+    "index.rst",
```

Comment:
Shouldn't these be excluded in pre-commit? Feels a bit weird to exclude them here.

Reply:
Hm, doesn't it make more sense to let this accept a `files` argument? That's how most of the linting, formatting, etc. tools work

Reply:
It does the job, we won't use it that much (when we modify the old changelog), it's not perfect but I don't see the point in refactoring it to perfection right now as it's blocking 2.14. Let's refactor it if we're going to keep the 
```
* blabla

  Closes #5445345
```  
format in #6688. Right now I think ``* blabla (#123456789)`` with 123456789 being the PR number and not the issue number would be better, which means we would delete this.

Reply:
Also see #6781, if we auto-generate the changelog we won't need it either